src/discussions/discussions-home/DiscussionSidebar.test.jsx

```diff
@@ -50,4 +69,22 @@ describe('DiscussionSidebar', () => {
     const element = await screen.findByTestId('sidebar');
     expect(element).toHaveClass('d-none');
   });
+
+  test('User with some topics should be redirected to "My Topics"', async () => {
+    const myOwnthreads = Factory.build('threadsResult', {}, {});
+    axiosMock.onGet(threadsApiUrl)
+      .reply(200, myOwnthreads);
```

Comment:
Here instead of returning a static response, you can use a function that returns a list of posts by a single author or multiple authors based on the username param.